### ISSUE TITLE
Add private function calls check to make sure there isn't one being called outside of the module it was defined 

### ISSUE THREAD

**Issue by Eric-Mendes** [astral-sh/ruff#9138]
I have drafted a [pre-commit hook](https://github.com/Eric-Mendes/private-calls-pre-commit) to make sure there isn't a private function being called outside of the module it was defined. Would it make sense as an optional linting rule? If so I'd be happy to contribute with code.

**Comment by charliermarsh** [comment]
Is this similar to [`private-member-access`](https://docs.astral.sh/ruff/rules/private-member-access/)?

**Comment by Eric-Mendes** [comment]
Perhaps it could be an extension to that rule, @charliermarsh. My code doesn't check for class private member access. It checks for module private member access. Here's an example of something that should raise an exception:
```python
# a.py
def _foo(*args, **kwargs):
    return "Hello World"
```

```python
# b.py
import a

print(a._foo())  # Raises an exception
```

```python
# c.py
from a import _foo

print(_foo())  # Also raises an exception
```

Have I misunderstood the rule you've mentioned, or am I really onto something here?

**Comment by charliermarsh** [comment]
I believe this would be closed by https://github.com/astral-sh/ruff/pull/5920.

**Comment by Eric-Mendes** [comment]
Indeed, @charliermarsh. Thanks for pointing that out, I didn't see that issue before.
